src/parse.rs

```diff
@@ -49,6 +49,58 @@ impl Error for ParseError {
     }
 }
 
+impl<'a> MimeIter<'a> {
+    /// A new iterator over mimes or media types
+    pub fn new(s: &'a str) -> Self {
+        Self {
+            pos: 0,
+            source: s,
+        }
+    }
+}
+
+impl<'a> Iterator for MimeIter<'a> {
+    type Item = Result<Mime, &'a str>;
+
+    fn next(&mut self) -> Option<Self::Item> {
+        let start = self.pos;
+        let len = self.source.bytes().len();
+
+        if start >= len {
+            return None
+        }
+
+        match parse(&self.source[start ..len]) {
+            Ok(value) => {
+                self.pos = len;
+                Some(Ok(value))
+            }
+            Err(ParseError::InvalidToken { pos, .. }) => {
+                if pos == 0 {
+                    self.pos += 1;
+                    return self.next()
```

Comment:
So I understand, if the source is a bunch of bogus bytes, this will try to parse over and over, moving one byte over?

Reply:
This follows the behavior of `parse()` and only deals with invalid tokens. If there is an invalid token, it tries to parse the string after or before (below). If the invalid token is the first character (such as in `, application/json`), `parse` immediately returns this position and yes, it moves one byte over.

If there are a bunch of random bytes without `/`, `parse` returns another error and this match immediately ends the iterator, for example, so it does not parse over and over the same string (minus one byte at a time).